File: descend_calc/descend_calc.cs
using System;

public class Program
{
    private static int GetInput(string msg)
    {
        while (true)
        {
            Console.Write(msg);
            string input = Console.ReadLine();
            int x = 0;
            if (Int32.TryParse(input, out x)) return x;
        }
    }

    public static void Main(string[] args)
    {
        int currentAltitude = GetInput("Current altitude: ");
        int airportAltitude = GetInput("Airport altitude: ");

        float altitudeToLose = ((currentAltitude - airportAltitude) / 100.0F) / 3.0F;

        int currentSpeed = GetInput("Current speed: ");
        int avgDscSpeed = GetInput("Average descend speed: ");

        float speedKnots = (currentSpeed - avgDscSpeed) / 10.0F;

        float windSpeed = GetInput("Wind speed: ") / 10.0F;
        int tailWind = GetInput("Tail or head wind [1 for tail | 0 for head]: ");

        float descendDistance = altitudeToLose + speedKnots + (tailWind == 1 ? windSpeed : -windSpeed);

        Console.WriteLine("------------------------------");
        Console.WriteLine("Altitude to lose nm: " + altitudeToLose);
        Console.WriteLine("Speed nm: " + speedKnots);
        Console.WriteLine("Wind  nm: " + windSpeed);
        Console.WriteLine("Total distance: " + descendDistance);
    }
}

File: descend_calc/descend_calc.vb
Imports System

Public Class Program
    Private Shared Function GetInput(msg As String) As Integer
        While True
            Console.Write(msg)
            Dim input As String = Console.ReadLine()
            Dim x As Integer
            If Int32.TryParse(input, x) Then
                Return x
            End If
        End While
    End Function

    Public Shared Sub Main(args As String())
        Dim currentAltitude As Integer = GetInput("Current altitude: ")
        Dim airportAltitude As Integer = GetInput("Airport altitude: ")

        Dim altitudeToLose As Single = ((currentAltitude - airportAltitude) / 100.0F) / 3.0F

        Dim currentSpeed As Integer = GetInput("Current speed: ")
        Dim avgDscSpeed As Integer = GetInput("Average descend speed: ")

        Dim speedKnots As Single = (currentSpeed - avgDscSpeed) / 10.0F

        Dim windSpeed As Single = GetInput("Wind speed: ") / 10.0F
        Dim tailWind As Integer = GetInput("Tail or head wind [1 for tail | 0 for head]: ")

        Dim descendDistance As Single = altitudeToLose + speedKnots + If(tailWind = 1, windSpeed, -windSpeed)

        Console.WriteLine("------------------------------")
        Console.WriteLine("Altitude to lose nm: " & altitudeToLose)
        Console.WriteLine("Speed nm: " & speedKnots)
        Console.WriteLine("Wind  nm: " & windSpeed)
        Console.WriteLine("Total distance: " & descendDistance)
    End Sub

End Class
